src/lib/coda_lib/coda_lib.ml

```diff
@@ -618,13 +618,17 @@ let best_path t =
     Transition_frontier.(root tf |> Breadcrumb.state_hash)
     (Transition_frontier.hash_path tf bt)
 
-let best_chain t =
+let best_chain ?max_length t =
   let open Option.Let_syntax in
   let%map frontier =
     Broadcast_pipe.Reader.peek t.components.transition_frontier
   in
-  Transition_frontier.root frontier
-  :: Transition_frontier.best_tip_path frontier
+  let best_tip_path = Transition_frontier.best_tip_path frontier in
+  match max_length with
+  | Some max_length when max_length <= List.length best_tip_path ->
+      best_tip_path
```

Comment:
did you want to take a sublist here?

Reply:
I've added a comment to clarify (and also the missing `?max_length` argument to `best_tip_path`).